server/routerlicious/packages/services-shared/src/runner.ts

```diff
@@ -110,6 +114,9 @@ export function runService<T extends IResources>(
 			process.exit(0);
 		},
 		async (error) => {
+			if (error.uncaughtException) {
+				runnerMetric.setProperty(CommonProperties.isUncaughtException, true);
```

Comment:
I know I mentioned it could be a boolean, but on second thought, would it be better to have a more genericProperty, like "restartReason" or something similar, and then have the value be "uncaughtException"?

That would help since we could also have a "SIGTERM", "caughtException" and other restart reasons.

Reply:
Hey Henrique, I was also considering something generic like "restartReason" but thought it would be simpler to just distinguish between caught and uncaught exceptions. Other reasons like sigterm can cause restarts but we dont have a way to identify that yet. 
However thinking more about it now, I think there's no harm in making it more generic as you suggested. I will update it now.